core/ledger/kvledger/txmgmt/statedb/stateleveldb/stateleveldb.go

```diff
@@ -220,6 +221,41 @@ func (vdb *versionedDB) GetFullScanIterator(skipNamespace func(string) bool) (st
 	return newFullDBScanner(vdb.db, skipNamespace)
 }
 
+// ImportState implements method in VersionedDB interface. The function is expected to be used
+// for importing the state from a previously snapshotted state. The parameter itr provides access to
+// the snapshotted state.
+func (vdb *versionedDB) ImportState(itr statedb.FullScanIterator, dbValueFormat byte) error {
+	if dbValueFormat != fullScanIteratorValueFormat {
+		return errors.Errorf("value format [%x] not supported. Expected value format [%x]",
+			dbValueFormat, fullScanIteratorValueFormat)
+	}
+	dbBatch := vdb.db.NewUpdateBatch()
```

Comment:
When we reset the leveldb batch, would it be better to expose leveldb `batch.Reset()` to reuse the allocated memory? If I understood the leveldb implementation of `batch.grow()`, if we use `batch.Reset()`, we might not use more than 4 MB (i.e., `maxDataImportBatchSIze`) in total. 

This is not particular to this PR as we create new batches in other places too. I can also take this up and make changes in all places at once if you agree.

Reply:
It is already exposed, as I had used the levedb Batch as an embedded field. Yes, it should be possible to reuse the buffer but not with any guarantee. The challenge being how we compute the batch size outside of this. Because, leveldb internally add additional bytes (between 3 and 11) per kv. Guaranteed reuse can come only at the cost of some ugly looking code at this layer, where we also account for additional max possible bytes per kv. But yes, in general, the reuse frequency may increase with the reset.